core/src/main/java/org/apache/carbondata/core/cache/CarbonLRUCache.java

```diff
@@ -67,6 +67,13 @@ public CarbonLRUCache(String propertyName, String defaultPropertyName) {
     } catch (NumberFormatException e) {
       lruCacheMemorySize = Integer.parseInt(defaultPropertyName);
     }
+
+    //if lru cache is bigger than jvm max memory then set to default size
+    if (isBeyondMaxMemory()){
+      LOGGER.info("LRU cache " + lruCacheMemorySize + "MB is beyond jvm max size, use default " + defaultPropertyName +" MB");
+      lruCacheMemorySize = Integer.parseInt(defaultPropertyName);
```

Comment:
So are you sure that the default value will not exceed the JVM heap size here? I think it may be a problem here!

Reply:
what we use for default is "-1"